### PR TITLE
[compiler-rt][AArch64][Android] Use getauxval on Android.

### PR DESCRIPTION
**Pull request by DanielKristofKiss**
__getauxval is a libgcc function that doesn't exist on Android.

### CODE REVIEW

**Comment by DanielKristofKiss** [comment]
Calling getauxval on Android is fine since:
https://android-review.googlesource.com/c/platform/bionic/+/836525

The [original](https://reviews.llvm.org/D154045#inline-1548473) motivation is broken as compiler-rt does call getauxval a lot from other places as of today.

File: compiler-rt/lib/builtins/aarch64/sme-abi-init.c [hunk 1]
```diff
@@ -20,11 +21,15 @@ _Bool __aarch64_has_sme_and_tpidr2_el0;
...
+#define GETAUXVAL(x) getauxval(x)
+#else
 extern unsigned long int __getauxval (unsigned long int);
+#define GETAUXVAL(x) __getauxval(x)
+#endif
 
-static _Bool has_sme(void) {
-  return __getauxval(AT_HWCAP2) & HWCAP2_SME;
-}
+static _Bool has_sme(void) { return GETAUXVAL(AT_HWCAP2) & HWCAP2_SME; }
```
Comment (compnerd) [inline compiler-rt/lib/builtins/aarch64/sme-abi-init.c:32]: I think that I would rather that we inline the use of `__getauxval`. `getauxval` is a GNU "standard" interface, but was introduced in glibc only in 2.16. I don't know if anyone is targeting an older glibc, but we can do something like:

```suggestion
static _Bool has_sme(void) {
#if defined(__GLIBC__) && !__GLIBC_PREREQ(2,16)
  unsigned long int auxval = __getauxval(AT_HWCAP2);
#else
  unsigned long int auxval = getauxval(AT_HWCAP2);
#endif
  return auxval & HWCAP2_SME;
}
```

Comment (compnerd) [reply compiler-rt/lib/builtins/aarch64/sme-abi-init.c]: It avoids the macros and allows us to have a single platform agnostic implementation. Over time, we would drop the use of the underscore variant of the function.

**Review by sdesmalen-arm — APPROVED** [review]
Sorry for the delay, I didn't realise this PR was still waiting for review.

**Pull request merged**